rust/agama-lib/src/questions/model.rs

```diff
@@ -0,0 +1,63 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Serialize, Deserialize, utoipa::ToSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct Question {
+    pub generic: GenericQuestion,
+    pub with_password: Option<QuestionWithPassword>,
+}
+
+/// Facade of agama_lib::questions::GenericQuestion
+/// For fields details see it.
+/// Reason why it does not use directly GenericQuestion from lib
+/// is that it contain both question and answer. It works for dbus
+/// API which has both as attributes, but web API separate
+/// question and its answer. So here it is split into GenericQuestion
+/// and GenericAnswer
```

Comment:
Perhaps we should consider refactoring the `GenericQuestion` from lib to be a structure composed of question and answer structs from this module. This is just an idea.